docs/api/cypress-api/custom-queries.mdx

```diff
@@ -306,6 +311,70 @@ be executed later (which is not idempotent!). `cy.now()` lets us directly call
 the original `get` query function, without Cypress' chaining and command queue
 logic wrapped around it.
 
+### Overwriting Existing Queries
+
+You can also modify the behavior of existing Cypress queries. This is useful to
+extend the functionality of builtin commands.
+
+:::caution
+
+`Cypress.Commands.overwriteQuery` can only overwrite queries, not other
+commands. If you want to modify the behavior of a non-query command, you'll need
+to use [`Cypress.Commands.overwrite`](/api/cypress-api/custom-commands) instead.
+
+:::
+
+```js
+Cypress.Commands.overwriteQuery('get', function (originalFn, ...args) {
+  console.log('get called with args:', args)
+
+  const innerFn = originalFn.apply(this, args)
+
+  return (subject) => {
+    console.log('get inner function called with subject:', subject)
+
+    return innerFn(subject)
+  }
+})
+```
+
+The `originalFn` is exactly the function originally passed to
+`Cypress.Commands.addQuery` - it is a **function that returns a function.** This
+gives you access to both the outer arguments (before you call `originalFn`) and
+the inner function (the return value of `originalFn`), giving you a great deal
+of control over how the query executes.
+
+Remember that query functions rely on `this` - when you invoke `originalFn`, be
+sure to use `.call` or `.apply`.
+
+#### Adding alias support to `.contains()`
+
```

Comment:
I think it'd be nice to explain this code segment briefly:

```suggestion

In this example, `cy.contains()` is extended to support querying for aliased subjects, like `cy.contains('@foo')`.

```

Reply:
Good call.